goal_src/jak2/engine/game/task/task-arrow.gc

```diff
@@ -5,5 +5,337 @@
+         (set! (-> self map-icon) a2-1)
+         (set! v0-2 (minimap-method-12 *minimap* self a2-1 (the-as int #f) (the-as vector #t) 0))
+         )
+       (set! (-> self minimap) (the-as connection-minimap v0-2))
+       v0-2
+       )
+      (('die)
+       (go-virtual die)
+       )
+      )
+    )
+  :code (the-as (function none :behavior task-arrow) sleep-code)
+  :post (behavior ()
+    (draw-arrow self)
+    (none)
+    )
+  )
+
+(defstate leave (task-arrow)
+  :virtual #t
+  :code (behavior ()
+    (set! (-> self moving) #t)
+    (while (-> self moving)
+      (suspend)
+      )
+    (process-drawable-method-17 self)
+    (none)
+    )
+  :post (behavior ()
+    (draw-arrow self)
+    (none)
+    )
+  )
+
+(defstate die (task-arrow)
+  :virtual #t
+  :code (behavior ()
+    (process-drawable-method-17 self)
+    (none)
+    )
+  )
+
+;; WARN: Return type mismatch object vs task-arrow.
+(defbehavior task-arrow-init-by-other task-arrow ((arg0 task-arrow-params))
+  (set! (-> self flags) (-> arg0 flags))
+  (set! (-> self map-icon) (-> arg0 map-icon))
+  (set! (-> self root) (new 'process 'trsqv))
+  (set! (-> self pos quad) (-> arg0 pos quad))
+  (if (logtest? (-> self flags) (task-arrow-flags task-arrow-flag-03))
+      (task-arrow-method-23 self (-> self pos))
+      )
+  (set! (-> self root trans quad) (-> self pos quad))
+  (quaternion-copy! (-> self root quat) (-> arg0 quat))
+  (quaternion-copy! (-> self base-quat) (-> arg0 quat))
+  (set-vector! (-> self root scale) 1.0 1.0 1.0 1.0)
+  (set! (-> self hud-dist) (the-as handle #f))
+  (set! (-> self max-dist) 0.0)
+  (set! (-> self theta) 0.0)
+  (set! (-> self phi) 0.0)
+  (set! (-> self minimap)
+        (minimap-method-12 *minimap* self (-> arg0 map-icon) (the-as int #f) (the-as vector #t) 0)
+        )
+  (cond
+    ((not (logtest? (-> self flags) (task-arrow-flags task-arrow-flag-02)))
+     (set! (-> self part) (create-launch-control (-> *part-group-id-table* 78) self))
+     )
+    (else
+      (initialize-skeleton
+        self
+        (the-as skeleton-group (art-group-get-by-name *level* "skel-task-arrow" (the-as (pointer uint32) #f)))
+        (the-as pair 0)
+        )
+      (set! (-> self draw lod-set lod 0 dist) 1228800.0)
+      (set-vector! (-> self draw color-mult) 0.5 0.5 0.0 1.0)
+      (set-vector! (-> self draw color-emissive) 0.5 0.5 0.0 1.0)
+      )
+    )
+  (set! (-> self rod-of-god-scale) 0.0)
+  (set! (-> self moving) #f)
+  (logclear! (-> self mask) (process-mask actor-pause movie))
+  (process-entity-status! self (entity-perm-status no-kill) #t)
+  (set! (-> self event-hook) (-> (method-of-object self idle) event))
+  (the-as task-arrow (go-virtual idle))
+  )
+
+;; WARN: Return type mismatch process vs none.
+(defun task-arrow-spawn ((arg0 task-arrow-params) (arg1 task-arrow))
+  (let* ((gp-0 (the-as process #f))
+         (s3-0 (get-process *default-dead-pool* task-arrow #x4000))
+         (v1-1
+           (when s3-0
```

Comment:
I'd also expect this function to look a little nicer on the current version of the decompiler.